### PR TITLE
Auto doc generation

### PR DESCRIPTION
**Pull request by zain-sohail**
Generating documentation on github pages. More details in #235

### CODE REVIEW

File: .github/workflows/pylint.yml [hunk 1]
```diff
@@ -3,7 +3,9 @@
 
 name: linting
 
-on: [push]
+on:
+  push: 
+    branches: main
```
Comment (zain-sohail) [inline .github/workflows/pylint.yml:8]: only there so we dont bother with lint checks till the end.

File: docs/tutorial/1 - Binning fake data.rst [hunk 2]
```diff
@@ -64,33 +64,33 @@ Generate Fake Data
       <tbody>
         <tr>
           <th>0</th>
-          <td>1.805042</td>
-          <td>-0.363059</td>
-          <td>-0.220710</td>
+          <td>-0.073557</td>
```
[the remark this answers is not in the document]
Comment (rettigl) [reply docs/tutorial/1 - Binning fake data.rst:67]: That's great, but the pictures don't appear in the read-the-doc output :-(

[the remark this answers is not in the document]
Comment (rettigl) [reply docs/tutorial/1 - Binning fake data.rst:67]: Maybe just the spaces in the filename are a problem

**Comment by rettigl** [comment]
It should work now, but will probably take ages.

**Comment by rettigl** [comment]
Yes, this does not work, the drive is not large enough...

File: docs/tutorial/1 - Binning fake data.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (zain-sohail) [reply docs/tutorial/1 - Binning fake data.rst:67]: Yes I believe so

**Comment by zain-sohail** [comment]
> Yes, this does not work, the drive is not large enough...

By default the runners come with 25-29 GB of storage so I wonder why. Maybe the zipped file + unzipped file combined is too big.

I see that the unzipped file is approximately 13 GB so if we can somehow directly unzip, maybe it works. Or take a subset of the whole data.

**Comment by zain-sohail** [comment]
> It should work now, but will probably take ages.

I think that won't be a big deal since changes to tutorials happen seldom

**Comment by zain-sohail** [comment]
So there is another way to go about auto generated of notebooks using 
https://nbsphinx.readthedocs.io/en/0.9.3/ 
or 
https://myst-nb.readthedocs.io/en/latest/
and that moves the notebook execution and build process to readthedocs. But the size issue still would occur as readthedocs probably doesnt have enough resources.

I don't like the manual rst (execute ipynb, create rsts and push) approach because that's very easy to not keep updated. Hence an automated way is best. Hence, I am still considering options.

**Comment by rettigl** [comment]
This looks much better now, but still it does not show in the docs. Not sure why.

**Comment by zain-sohail** [comment]
> This looks much better now, but still it does not show in the docs. Not sure why.

I think just the files having spaces in the name issue. I will compile docs locally and see if that's the case.

Can you look into this: https://github.com/OpenCOMPES/sed/actions/runs/6693101434/job/18183570962#step:7:448 ? The notebook 2 is not fully executing

**Comment by rettigl** [comment]
Oh, yes seems like something I have to fix.
Regarding the workflow, I would suggest to actually download and unpack the test data outside the notebook to not have this output inside the generated example file. Also, can we make sure the example 2 is executed before example 3?

**Comment by zain-sohail** [comment]
> > I think just the files having spaces in the name issue. I will compile docs locally and see if that's the case.
> 
> The files are not linked in the index

Yes but that doesn't address the images not appearing. So indeed without spaces, it fixes the problem. I will push a commit that removes spaces during the workflow only

**Comment by zain-sohail** [comment]
> Oh, yes seems like something I have to fix. Regarding the workflow, I would suggest to actually download and unpack the test data outside the notebook to not have this output inside the generated example file. Also, can we make sure the example 2 is executed before example 3?

Can't we just suppress the output of that cell in the notebook? 
Regarding the execution order, not sure how but I will take a look. Is it strictly necessary? I saw that 3 executed just fine.

**Comment by rettigl** [comment]
> > Oh, yes seems like something I have to fix. Regarding the workflow, I would suggest to actually download and unpack the test data outside the notebook to not have this output inside the generated example file. Also, can we make sure the example 2 is executed before example 3?
> 
> Can't we just suppress the output of that cell in the notebook? Regarding the execution order, not sure how but I will take a look. Is it strictly necessary? I saw that 3 executed just fine.

Yes, but only because sed_confg is accidentally there. But we can also leave it like this, then they don't depend on each other

**Comment by rettigl** [comment]
I just renamed the notebooks now, I think this is the better fix. Seems to work now, except that all bokeh figures are missing.

**Comment by rettigl** [comment]
https://github.com/Chilipp/sphinx-nbexamples/issues/10

**Comment by zain-sohail** [comment]
> I just renamed the notebooks now, I think this is the better fix. Seems to work now, except that all bokeh figures are missing.

I assumed we wanted to have the names in that format but definitely this is a better fix.

Seems like for having bokeh, just executing the notebooks first and then using nbconvert should work

**Comment by zain-sohail** [comment]
> Seems to work now, except that all bokeh figures are missing.

Actually, I see the bokeh figures on readthedocs. Or did you mean with sliders etc?

**Comment by rettigl** [comment]
> > Seems to work now, except that all bokeh figures are missing.
> 
> Actually, I see the bokeh figures on readthedocs. Or did you mean with sliders etc?

E.g. the cell just before 2.2. should output the event histrograms. Its's empty for me... I also did not get it to work locally. Conversion to html does work, but not to rst.

**Comment by rettigl** [comment]
Maybe this is related: 
https://github.com/jupyter/nbconvert/issues/1573

**Pull request merged**